### PR TITLE
perf: move all devDependencies to root package.json, prettier all files

### PR DESCRIPTION
**Pull request by JounQin**
The  devDependencies are all redundant added and installed, this PR makes them simple.

### CODE REVIEW

**Comment by JounQin** [comment]
Additionally, I want to run `prettier` for all possible files including `.md` and `.mdx`.

https://github.com/mdx-js/mdx/issues/844

What do you think?

@mdx-js/core

**Comment by JounQin** [comment]
Hey guys, finally, the all tests pass! Please help to review this PR which is very important internal improvement!

cc @mdx-js/core @mdx-js/maintainers

**Comment by timneutkens** [comment]
In general it'd be better to do the prettier change in a separate PR. But overall I think this is fine (didn't review line by line).

**Comment by wooorm** [comment]
Same as Tim; looks good and would suggest, for the future, to split PRs to do one thing (especially because the prettier changes are so big, making it hard to review!)

**Comment by JounQin** [comment]
@timneutkens @wooorm Thx for your great suggestions, I'll try to do better next time.

Would you like to approve this PR?

**Comment by JounQin** [comment]
I'm going to merge this PR without your approves temporarily. Feel free to point out my faults and I'll fix them ASAP.

File: packages/runtime/package.json [hunk 1]
```diff
@@ -23,41 +14,29 @@
...
-  "main": "dist/index.js",
-  "module": "dist/index.es.js",
-  "scripts": {
-    "build": "microbundle --format es,cjs --external react,buble-jsx-only,@mdx-js/mdx,@mdx-js/react --jsx React.createElement --no-sourcemap",
-    "clean": "rimraf dist",
-    "prepublish": "yarn clean && yarn build",
-    "test": "jest"
-  },
+  "license": "MIT",
+  "main": "dist/cjs.js",
```
Comment (ChristianMurphy) [inline packages/runtime/package.json]: is this change intended?
Would a build update be needed to support this?
see https://github.com/mdx-js/mdx/issues/876

File: package.json [hunk 2]
```diff
@@ -35,21 +20,107 @@
     "examples/shortcodes",
     "examples/terminal"
   ],
+  "scripts": {
+    "bootstrap": "yarn && lerna bootstrap && lerna link --force-local",
+    "build": "r -f cjs,esm -o dist --exclude 'babel-plugin-*,create-mdx,*loader,mdx,parcel-plugin-mdx,remark-*,runtime,*util'",
```
Comment (ChristianMurphy) [inline package.json]: Is this intended to be rollup or another command?

File: package.json [hunk 3]
```diff
@@ -35,21 +20,107 @@
...
+    "clean": "lerna exec \"rm -rf node_modules\"",
```
Comment (ChristianMurphy) [inline package.json]: should clean also remove `dist` folders?

File: packages/runtime/package.json (hunk 1 above)
Comment (JounQin) [reply packages/runtime/package.json]: @ChristianMurphy Thx for point out, I'm working on it.

File: package.json (hunk 3 above)
Comment (JounQin) [reply package.json]: I didn't change it, it's just sort by `prettier-plugin-pkg`.

File: package.json (hunk 2 above)
Comment (JounQin) [reply package.json]: It's a wrapper [@pkgr/rollup](https://github.com/rx-ts/pkgr/tree/master/packages/rollup) which is mush simple to use.

Comment (ChristianMurphy) [reply package.json]: gotcha, good to know, it confused me for a bit because some shells have [`r` as a command that repeats the previous command](https://linux.die.net/man/1/zshbuiltins)
